src/horust/healthcheck.rs

```diff
@@ -181,4 +237,21 @@ file-path = "{}""#,
         assert!(healthchecks(&healthiness));
         Ok(())
     }
+
+    #[test]
+    fn test_http_healthiness_check() {
+        let loopback = Ipv4Addr::new(127, 0, 0, 1);
+        let socket = SocketAddrV4::new(loopback, 22222);
+        let listener = TcpListener::bind(socket).unwrap();
+        let local_addr = listener.local_addr().unwrap();
+        let endpoint = format!("http://127.0.0.1:{}", local_addr.port());
+        thread::spawn(move || {
+            start_server(socket);
+        });
+        let healthiness = Healthiness {
+            file_path: None,
+            http_endpoint: Some(endpoint),
+        };
+        assert!(!healthchecks(&healthiness));
```

Comment:
This is almost it! So one thing, before running the webserver you should assert the healthcheck is failing. Then you start the webserver and you assert that the healthcheck is passing. 

In this case actually, it should work with `assert!(healthchecks(&healthiness));`. If that is not the case, maybe there is a bug in the `healthchecks` function.